### PR TITLE
Wizard Spell: Doppelganger

### PR DESCRIPTION
**Pull request by Shadowmech88**
Adds a new spell, Doppelganger. It costs 10 spell points, half that of a standard spell.
Each use spawns a simple wizard mob with the same name and appearance as the wizard who summoned it. It is hostile to everyone but its summoner. By default it will only melee attack, but can use magic missile if the spell is upgraded.
The doppelgangers do not expire, nor is there a limit on how many the wizard may summon.
The spell has a default cooldown of 30 seconds, which can be decreased to 10 seconds for 20 spell points.

I'm going to wait until #16279 is merged to complete this because .dmi conflicts are not fun.

:cl:
 * rscadd: Added a new spell, Doppelganger. It creates a body double of the wizard that can be used for misdirection or reinforcements. If the spell is upgraded, the doppelganger becomes able to fire magic missiles.

### CODE REVIEW

**Comment by Aurx** [comment]
No limit and magic missile makes me think you'll see wizzymen holing up somewhere and just amassing a giant army of unstunable stun-spamming clones.  Is that intended player behavior?

**Comment by Shadowmech88** [comment]
He could amass an army, but he couldn't guide it. The doppelgangers just wander around, they don't follow the wizard.

**Comment by Cyadox** [comment]
I'd be more concerned with it being like spider mobs, where they amass to the point that only a bomb will kill them. There's no limit to what a wizard could do if he just holed them up in a room like virology, etc.

I'm not saying that's a bad thing. Personally more excuses to legitimately use bombs is fine in my books, but I can see some butthurt over spamming the spell in a closed room and essentially making it impossible by conventional means to kill the wizard when he's surrounded by look-alikes.

**Comment by Hugh-G-Rection** [comment]
Staff of Necromancy already exists and lets you create infinite armies as well, a more dangerous one in my opinion.

**Comment by Exxion** [comment]
NEW WIZARD STRAT:
1. Steal the hand teleporter
2. Find or build a secret room somewhere people won't find it
3. Sit in there for half an hour spawning these exactly as they go off cooldown
4. Open portal when ready to flood the station

**Comment by Hugh-G-Rection** [comment]
Same thing I've done on the satellite a dozen times, but the issue here is you wouldn't be able to get them into the portal rapidly en-masse like spiders/russians because they aren't hostile towards you. Portal doesn't send things that are in the same tile when it opens, only things that move into the space afterwards, OR things that are created or dropped on it while it's open, which is why the goat trick works by opening stacked crates, or necromancy on stacked corpses, or baguloose appearing on the other side and not killing you.

On second thought I guess you could put the portal in a funnel and drag a monkeyman onto the opposite side for bait? Could also stuff them all in a locker but good luck with that.

**Comment by Aurx** [comment]
Any idiot can make a conveyor belt.

**Comment by Lokiamis** [comment]
You could make them target each other if they haven't attacked anyone else in a while, or something to that effect.

File: code/modules/spells/aoe_turf/conjure/doppelganger.dm [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+	level_max = list(Sp_TOTAL = 3, Sp_SPEED = 2, Sp_POWER = 1)
+	charge_max = 300
+	cooldown_reduc = 100
+	cooldown_min = 100
+	invocation = "MY O'N CLO'N"
+	invocation_type = SpI_SHOUT
+	spell_flags = NEEDSCLOTHES
+	hud_state = "wiz_snakes"
+
+var/list/doppelgangers
```
Comment (PJB3005) [inline code/modules/spells/aoe_turf/conjure/doppelganger.dm]: Just initialize this list in the declaration.

**Review by PJB3005 — CHANGES_REQUESTED** [review]

**Review by PJB3005 — APPROVED** [review]

**Comment by Shadowmech88** [comment]
Git told me I had 15,000 additions and deletions on Bagelstation and wouldn't let me change branches until I committed, and now I don't know how to get rid of them.

**Comment by PJB3005** [comment]
Er, do you just want to completely remove bagel from the diff or do you have some changes to the map?

If you want to just remove it, get the version from the main repo and commit that.

**Comment by Shadowmech88** [comment]
It wouldn't let me directly download it, I had to save it as a text file and then re-save it as a .dmm with Notepad++.
When I tried re-committing it, it said I had no changes to commit.

I think the issue is something to do with line endings, and it may not be possible for me to save the repo's copy of the map without automatically doing those changes to every single line.

**Comment by Shadowmech88** [comment]
Also, now literally every branch I check out says I have changes to bagelstation, and won't let me switch branches until I commit or stash the changes.

**Comment by ghost** [comment]
Yeah, even if someone tries to swarm meme, just walk away and come back with flamethrowers to hose the place down or just weld it up and burn it out with plasoxy. I don't know the AI exactly but it should have enough of the usual acquisition delay that anyone not asleep at the keyboard should be able to start booking it if they stumble on one.

@Shadowmech88 Can you put an arbitrary global cap at like 40? 50? and have it spit back a feedback message to the caster(s) when they try to make one over the cap? I know the optimizations have helped since spider apocalypses killing the server but why tempt fate.

**Comment by Exxion** [comment]
That was completely a joke. AIs with no interaction with each other have no problems in numbers that people could feasibly get to by making one mob at a time.

**Comment by ghost** [comment]
Alrighty then, I just remember the old chug from spider apocalypses and wanted to get ahead of it if say 2-3 wizards took and spammed it from different parts of the station.

**Comment by Shadowmech88** [comment]
The PR's fixed, though I still need to get Git sorted out.

**Pull request merged**